httpx/_client.py

```diff
@@ -456,16 +459,24 @@ def __init__(
 
         proxy_map = self.get_proxy_map(proxies, trust_env)
 
-        self.dispatch = self.init_dispatch(
+        if dispatch is not None:
+            warnings.warn(
```

Comment:
Do we want to make the `category` of these warnings `DeprecationWarning` (like #908)?